R/get_code.R

```diff
@@ -92,6 +93,42 @@ get_code.TealDataAbstract <- function(x, dataname = character(0), deparse = TRUE
   }
 }
 
+#' @rdname get_code
+#' @export
+#' @examples
+#'
+#' tdata1 <- teal_data()
+#' tdata1 <- within(tdata1, {
+#'   a <- 1
+#'   b <- a^5
+#'   c <- list(x = 2)
+#' })
+#' get_code(tdata1)
+#' get_code(tdata1, "a")
+#' get_code(tdata1, "b")
+#'
+#' tdata2 <- teal_data(x1 = iris, code = "x1 = iris")
+#' get_code(tdata2)
+#' get_code(is_reproducible(tdata2))
+#'
+get_code.teal_data <- function(x, dataname = character(0), deparse = TRUE, ...) {
+  code <- if (length(dataname) > 0) {
+    teal.code:::get_code_dependency(x@code, dataname)
```

Comment:
Oh, this is an issue. We shouldn't use internals from a different package, even from a hard dependency.
One way to get around it is to use `getFromNamespace` like we do in `teal` but it would be better to avoid it entirely, if possible.

I just noticed that `get_code_dependency` is not used in `teal.code` at all. Moving it to `teal_data` doesn't seem right. Maybe it should be exported?

Reply:
We had this conversation with @gogonzo some time ago about this. The `get_code_dependency` is intentionally not exported and not used anywhere, so that we can use this only here :) I guess we could export `get_code_dependency` but only if we improve some `poor sides of the implementation` in here https://github.com/insightsengineering/teal.code/issues/154

Currently `get_code_dependency()` assumes specific structure of `code_dependency()` that we think can be simplified.

Reply:
so far calling `get_code_dependency` by `getFromNamespace`. the same as we do for `format_expression` 313bfa40345159499eb360619bfaea99d48f9230

Reply:
get_code_dependency was moved to teal.data